### PR TITLE
Dark and Light Theme Toggle Functionality

### PR DESCRIPTION
**Pull request by stevenovak123**

### CODE REVIEW

**Comment by stevenovak123** [comment]
I ran the scripts in the frontend to lint and still they have shown up in the changes. Sorry for the difficulty caused.

**Comment by nirsht** [comment]
@stevenovak123 can you explain more in details what is the problem ?

**Comment by nirsht** [comment]
@stevenovak123 Can you also attach pics to here ?

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
+  }, []);
+
   return (
     <Router>
       <ScrollToTop />
-      <Container>
-        <GlobalStyle />
-        <PrivateRoutes />
-      </Container>
+      <ThemePreference.Provider value={{ currentTheme, setCurrentTheme }}>
```
Comment (nirsht) [inline frontend/src/App.tsx:43]: I rather seperate it to different file containing thr ThemeProvider

File: frontend/src/components/MainView/MainViewHeader.tsx [hunk 2]
```diff
@@ -91,43 +91,51 @@ const LogoValidkubeMobileContainer = styled(LogoModblieValidkube)`
...
-      </RightHeaderContainer>
-    </Container>
-    <StyledHr />
-  </TopContainer>
-);
+const MainViewHeader: React.FC = () => {
+  const theme = useContext(ThemePreference);
+
+  const togglerTheme = theme?.currentTheme === 'dark' ? 'light' : 'dark';
+  console.log(togglerTheme);
```
Comment (nirsht) [inline frontend/src/components/MainView/MainViewHeader.tsx:98]: Please remove that

**Review by nirsht — COMMENTED** [review]
Also please try to solve the changes from `""` to `''`, this is probably the formatter that you are using, try use prettier

**Comment by stevenovak123** [comment]
> @stevenovak123 can you explain more in details what is the problem ?

I use prettier as well but my config has them in ' ' instead of " ". I think it is due to the fact there is not a prettier rules file set in the project. if that is that it will automatically lint it to a default set.

File: frontend/src/App.tsx (hunk 1 above)
Comment (stevenovak123) [reply frontend/src/App.tsx:43]: I tried separating it and ran into many type errors and problems.

**Comment by stevenovak123** [comment]
Added the necessary changes as requested. separated the context logic and corrected the " ". Attaching the images of how the changes look. The theme switching button is located on the Navbar beside the github icon.
validkube
validkubeLightMode_

**Comment by nirsht** [comment]
Hi @stevenovak123 looks great, just to verify, the default theme is dark ?

I am AFK for few days, after that I will merge it and deploy it 🎉

**Review by nirsht — APPROVED** [review]

**Comment by stevenovak123** [comment]
> Hi @stevenovak123 looks great, just to verify, the default theme is dark ?
> 
> I am AFK for few days, after that I will merge it and deploy it 🎉

The default is set depending on the users' browser settings.

**Pull request merged**